core/src/main/java/org/commonjava/maven/ext/manip/impl/InfoManipulator.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2012 Red Hat, Inc. (jcasey@redhat.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.commonjava.maven.ext.manip.impl;
+
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.commonjava.maven.ext.manip.ManipulationException;
+import org.commonjava.maven.ext.manip.ManipulationManager;
+import org.commonjava.maven.ext.manip.ManipulationSession;
+import org.commonjava.maven.ext.manip.io.PomIO;
+import org.commonjava.maven.ext.manip.model.Project;
+import org.commonjava.maven.ext.manip.state.InfoState;
+import org.commonjava.maven.ext.manip.state.InfoState.GAV;
+
+import java.util.List;
+
+/**
+ * {@link Manipulator} implementation that gathers various information about the project.
+ * The information is intended to be saved into {@link ManipulationManager#RESULT_FILE}
+ */
+@Component( role = Manipulator.class, hint = "info-manipulator" ) public class InfoManipulator
+                extends AbstractNoopManipulator
+{
+    @Requirement private PomIO pomIO;
```

Comment:
If you're providing an afterManip functionality then why do we need another Manipulator - shouldn't the ProjectVersioningManip be responsible for storing the new GAV once its completed altering it after manipulation?